cli/cmdlineparser.cpp

```diff
@@ -1183,64 +1206,62 @@ void CmdLineParser::printHelp()
-    "                         For more information about library files, read the\n"
-    "                         manual.\n"
-    "    --max-configs=<limit>\n"
-    "                         Maximum number of configurations to check in a file\n"
-    "                         before skipping it. Default is '12'. If used together\n"
-    "                         with '--force', the last option is the one that is\n"
-    "                         effective.\n"
-    "    --max-ctu-depth=N    Max depth in whole program analysis. The default value\n"
-    "                         is 2. A larger value will mean more errors can be found\n"
-    "                         but also means the analysis will be slower.\n"
-    "    --output-file=<file> Write results to file, rather than standard error.\n"
-    "    --platform=<type>, --platform=<file>\n"
-    "                         Specifies platform specific types and sizes. The\n"
-    "                         available builtin platforms are:\n"
-    "                          * unix32\n"
-    "                                 32 bit unix variant\n"
-    "                          * unix64\n"
-    "                                 64 bit unix variant\n"
-    "                          * win32A\n"
-    "                                 32 bit Windows ASCII character encoding\n"
-    "                          * win32W\n"
-    "                                 32 bit Windows UNICODE character encoding\n"
-    "                          * win64\n"
-    "                                 64 bit Windows\n"
-    "                          * avr8\n"
-    "                                 8 bit AVR microcontrollers\n"
-    "                          * elbrus-e1cp\n"
-    "                                 Elbrus e1c+ architecture\n"
-    "                          * pic8\n"
-    "                                 8 bit PIC microcontrollers\n"
-    "                                 Baseline and mid-range architectures\n"
-    "                          * pic8-enhanced\n"
-    "                                 8 bit PIC microcontrollers\n"
-    "                                 Enhanced mid-range and high end (PIC18) architectures\n"
-    "                          * pic16\n"
-    "                                 16 bit PIC microcontrollers\n"
-    "                          * mips32\n"
-    "                                 32 bit MIPS microcontrollers\n"
-    "                          * native\n"
-    "                                 Type sizes of host system are assumed, but no\n"
-    "                                 further assumptions.\n"
-    "                          * unspecified\n"
-    "                                 Unknown type sizes\n"
-    "    --plist-output=<path>\n"
-    "                         Generate Clang-plist output files in folder.\n";
+        "    -l <load>            Specifies that no new threads should be started if\n"
```

Comment:
hmm.. somehow it seems not ideal to display this option in the help output when it's not available at all. if we include it in the help output I suggest that we indicate that it is not available.

Reply:
It is included in the manpage and the manual which are both static files.

The command-line should be treated like any API which is a fixed interface which will provide an error when something is not working/available.

Also some of this might be based on on checks in `config.h` which are based on the underlying system/compiler and might not even be controllable by the user which builds the binary (like not having process support on Windows).

We might be able to improve the documentation a bit but that should not be in the `--help` output which is just an overview and not an in-depth documentation. That's what the manpage is for.

Reply:
Yes the manual is static as you say. But in this case we can be dynamic. How about writing a "NOT AVAILABLE" at least?

Reply:
That would be quite subtle and re-introduce the usage of the preprocessor checks in the help which I also wanted to get rid of.

I guess we should use a struct to store information on each option and generate the help from that. That could also help with generating the manpage output so the information not duplicated. We could also add a note to the static documentation that the availability of that option depends on the platform or build options.

Reply:
> We could also add a note to the static documentation that the availability of that option depends on the platform or build options.

:+1: